ARM: dts: meson8m2: mxiii: Support Minix X8-H Plus

amondit opened the pull request:
Add DTS file to support Minix Neo X8-H Plus board.

Board is similar than the mxiii plus board but with 16GB eMMC
(I'm currently playing to add support for it) and as a AP6335
bt/wifi combo chip.

amondit committed:
ARM: dts: meson8m2: mxiii: Support Minix X8-H Plus

Add DTS file to support Minix Neo X8-H Plus board.

Board is similar than the mxiii plus board but with 16GB eMMC
(I'm currently playing to add support for it) and as a AP6335
bt/wifi combo chip.

amondit commented:
Hello, I'm not sure about the process for PRs, so I just opened it just to share some code I've been playing with. Feel free to disregard/play with it as you feel necessary. 

Also I wasn't sure if it made sense to base of the meson8m2-mxiii.dtsi file as it's another brand of board (or directly go from the base meson8m2.dtsi one), but there seems to be some similarities between the boards. It could even be argued to directly include the changes in the meson8m2-mxiii-plus.dts file as what I did until now should be valid for this board as well (but that may not be the case anymore once I start playing with emmc, leds, etc.)

Thanks a lot for your work on the platform anyway!

xdarklight commented:
Hi @amondit 

thank you for your contribution!

The "standard" way would be to send patches to the linux-amlogic mailing list.
I'll still go through your patch and give feeback - then we can also decide how to proceed.

My spare time this week is very limited so please apologize if you don't hear back from me until Sunday.

amondit commented:
Sure no problem, there's no rush. I have limited availability as well. Thanks already for all the time you took for this project!

amondit commented on arch/arm/boot/dts/Makefile:
```diff
@@ -347,6 +347,7 @@ dtb-$(CONFIG_MACH_MESON8) += \
 	meson8b-odroidc1.dtb \
 	meson8m2-mxiii.dtb \
 	meson8m2-mxiii-plus.dtb \
+	meson8m2-minix-neo-x8-h-plus.dts \
```
typo in the extension, should be dtb

xdarklight commented on arch/arm/boot/dts/meson8m2-minix-neo-x8-h-plus.dts:
```diff
@@ -0,0 +1,98 @@
+	pinctrl-names = "default";
+	uart-has-rtscts;
+
+    bluetooth {
```
the indentation here seems wrong

amondit replied:
4 spaces vs tab. Fixed

xdarklight commented on arch/arm/boot/dts/meson8m2-minix-neo-x8-h-plus.dts:
```diff
@@ -0,0 +1,98 @@
+	compatible = "minix,x8-h-plus", "amlogic,meson8m2";
+
+	aliases {
+		serial1 = &uart_A;
```
this shouldn't be necessary because we're not using this as user-exposed serial port but for communication with the Bluetooth module (so the serial port is handled on the kernel side and doesn't need this alias).

I probably have to clean up some of the existing .dts files which have this though

amondit replied:
Ok, it was leftover from copypaste from mxiii-plus dts. Removed in last commit

xdarklight commented on arch/arm/boot/dts/meson8m2-minix-neo-x8-h-plus.dts:
```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2018 Oleg Ivanov <balbes-150@yandex.ru>
+ * Copyright (c) 2018 Martin Blumenstingl <martin.blumenstingl@googlemail.com>
+ * Copyright (c) 2020 Arnaud Mondit <arnaud.mondit@gmail.com>
```
For this I prefer a copyright style similar to what we have in ``arch/arm64/boot/dts/amlogic/meson-gxl-s905w-tx3-mini.dts`` to make it clear that not all of us worked on that file

amondit replied:
Fixed that on the last commit, I hope it's better?

xdarklight commented:
overall this is looking good. I added a few small comments inline.

Also there's no patch for the dt-bindings similar to for example what Christian has done in 8b6f53d7b94b8d23860ae44a56b0af404a6c7e4c
This is a must-have (in a separate commit) for inclusion upstream

my main concern is that I haven't thought about how to add the (my own) .dtsi upstream yet

I'll take that as TODO for another bad-weather day.
when I have a patch ready I'll Cc you on the mail to get that .dtsi upstream.
then I'll ask you to submit the patch on the mailing list as Kevin only takes patches from there

amondit committed:
ARM: dts: meson8m2: Minix X8HP: Fix bluetooth

Set bluetooth UART speed so that the bluetooth module
has enough bandwidth for higher bitrates. Most seen
when doing a2dp streaming.

amondit committed:
dt-bindings: arm: amlogic: add support for the Minix Neo X8-H Plus

The Minix Neo X8-H Plus is a commercial Android device based on the
Amlogic N200 reference design using the S812-H chipset. Specs:

2GB DDR3 RAM
16GB eMMC storage
10/100/1000 Ethernet using Realtek RTL8211F
802.11 a/b/g/n/ac + BT 4.0 sdio wireless module (AP6335)
1x blue colour LED to indicate power
1x power button
1x reset button
HDMI 1.4 (4k@30p) video
2-channel audio input/output on 2x 3.5mm jacks
Optical S/PDIF audio output
3x USB 2.0 ports
1x USB OTG port
1x SD card slot
UART pins (internal)
IR Sensor
